### DISCUSSION TITLE
[question][solved]: how do I add to an array contents from another file

### DISCUSSION THREAD

**Discussion by luntik2012** [mikefarah/yq#1329] [Q&A] [answered]
src:
```yaml
_anchors_:
  - &_test 123
```

dst:
```yaml
_anchors_:
  - &_test 123
  - $_myVar |
    multilineCertFromFileLine0
    multilineCertFromFileLine1
    multilineCertFromFileLine2
```

**Comment by mikefarah** [comment, marked as answer]
Note: Please raise future how to questions under discussion please :)

```bash
yq '._anchors_ += loadstr("cert.txt") | (._anchors_[-1] anchor="_myVar")' data.yaml
```

Explanation:
- You need to append to the `_anchors_` array: `._anchors_ =+ (blah)`
- Load contents as a string from a file `loadstr("cert.txt")`
- Next add the anchor to the new element. We do this by selecting the last element in the array, and assign it an anchor: `(._anchors_[-1] anchor="_myVar")`

**Reply by luntik2012** [reply]
thank you
